hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerRatisServer.java

```diff
@@ -591,9 +591,10 @@ public void start() throws IOException {
   }
 
   public void stop() {
+    LOG.info("Stopping {} at port {}", this, port);
     try {
+      // Ratis will also close the state machine
       server.close();
-      omStateMachine.close();
```

Comment:
I faced same problem that omStateMachine when stopped, takeSnapshot() block for double buffer flush. Below changes may be requried for this PR, and need revert omStateMachine.close()

https://github.com/apache/ozone/pull/5857/commits/493323a2632a5f28f0486f775e50c1817de941b1

Reply:
> ... takeSnapshot() block for double buffer flush. ...

That is a good point.  Let's add a check in `takeSnapshot()`.

> ... need revert omStateMachine.close()

`RaftServer` will close `omStateMachine`.

Reply:
good logic to track and update last applied index in combination with notified index.